Documentation/rust/unit-testing.rst

```diff
@@ -0,0 +1,49 @@
+.. SPDX-License-Identifier: GPL-2.0
+
+Unit Testing
+============
+
+Like in any other Rust project it is possible to write and run unit tests and
+documentation tests in the kernel.
+
+Running Unit Tests
+------------------
+
+Unit tests in the kernel are identical to user-space Rust tests:
+
+.. code-block:: rust
+
+	#[cfg(test)]
+	mod tests {
+	    #[test]
+	    fn it_works() {
+	        let result = 2 + 2;
+	        assert_eq!(result, 4);
+	    }
+	}
+
+And can be run using the ``rusttest`` Make target:
+
+.. code-block:: bash
+
+	$ make LLVM=1 rusttest
+
+Running Documentation Tests
+---------------------------
```

Comment:
I'd probably personally consider doctests at least "unit-like": they seem to mostly be a mix between unit tests and "compile tests", and rarely would be integration tests. KUnit certainly has taken a rather broad definition of unit tests (and, of course, given we're using it for doctests, I'd argue they qualify), so I wouldn't object to including doctests on a page about unit testing.

That being said, I do think that just naming the file "testing" is probably the way to go anyway, unless there's a particular reason to have a separate page for integration or some other form of testing.

Reply:
I just pushed v2 changing the page's title to "Testing". I considered "Automated Testing" so it is clear that this section does not cover manual testing, but since the both of you agreed on the title I leaved it as "Testing".